directus-job-board-platform/index.md

```diff
@@ -0,0 +1,1126 @@
+
+- `id:` Autocomplete Input
+- `status`: input field
+
+## Creating relationships in Directus
+
+To link a new job with its creator:
+
+1. In the `job` collection, add a Many-to-One field named `employer`.
+2. Set the related collection to System > `directus_users`.
+3. Choose `first_name` as the Display template.
+
+This creates a relationship between the job and the admin who created it.
+
+In the `application` collection, add two Many-to-One fields:
+
+1. `user`: Links to the applicant
+2. `job`: Links to the job
+
+These fields connect each application to its corresponding applicant and job. Add 3 items in the `job` collection - [here's some sample data](https://raw.githubusercontent.com/preshenv/directus-solidstart-job-board/main/src/components/jobs.md).
+
+## Creating a New User Role
+In your access control settings, create a new role called `Job Applicant`. For the `application` collection, enable `create` and `read` permissions. Use custom rules for the `application` collection to ensure users can only read and update their own applications. Set a filter like: `user -> id Equals $CURRENT_USER.id`. This allows users to view all jobs, create new applications, and view or update only their own applications. 
+
+## Allowing Public Registration
+Open the public role, find `directus_users` under system collections, and then open custom permissions for the `create` operation. 
+1. In field permissions, enable only `first_name`, `last_name`, `email`, and `password` options. If you want users to provide other data at the time of registration, enable the respective field.
+2. In field validation, require `first_name`, `last_name`, `email`, and `password` to not be empty.
+3. In field presets, set the new user's `role` to the `Job Applicant` role id. 
+
+Give public view access to the `job` collection to allow users to see available jobs even when they are not logged in. Also, ensure the admin role retains full permissions across all collections.
+
+## Initializing the SolidStart.js project
+Create a new SolidStart project by running the command:
+
+```env
+ npm init solid@latest
+```
+Choose the **bare** template, enable Server-side rendering from the prompt, and Typescript in the prompts.
+
+## Setting up Directus instance
+In your SolidStart project's `src` directory, create a `lib` directory. Inside it, create a `directus.js` file:
+
+```jsx
+import { authentication, createDirectus, rest } from "@directus/sdk";
+
+export const PUBLIC_DIRECTUS_API_URL = import.meta.env
+  .VITE_PUBLIC_DIRECTUS_API_URL;
+
+function getDirectusInstance() {
+  const directus = createDirectus(PUBLIC_DIRECTUS_API_URL)
+    .with(
+      authentication("cookie", { credentials: "include", autoRefresh: true })
+    )
+    .with(rest({ credentials: "include" }));
+  return directus;
+}
+export default getDirectusInstance;
+```
+Add your Directus URL to the `.env` file:
+
+```env
+VITE_PUBLIC_DIRECTUS_API_URL='https://directus.example.com'
+```
```

Comment:
````suggestion
## Initializing a SolidStart.js project

Create a new SolidStart project by running the command:

```env
npm init solid@latest
```

Choose the **bare** template, enable Server-side rendering from the prompt, and Typescript in the prompts.

In your SolidStart project's `src` directory, create a `lib` directory. Inside it, create a `directus.js` file:

```jsx
import { authentication, createDirectus, rest } from "@directus/sdk";

export const PUBLIC_DIRECTUS_API_URL = import.meta.env
  .VITE_PUBLIC_DIRECTUS_API_URL;

function getDirectusInstance() {
  const directus = createDirectus(PUBLIC_DIRECTUS_API_URL)
    .with(
      authentication("cookie", { credentials: "include", autoRefresh: true })
    )
    .with(rest({ credentials: "include" }));
  return directus;
}
export default getDirectusInstance;
```

Add your Directus URL to the `.env` file:

```env
VITE_PUBLIC_DIRECTUS_API_URL='https://directus.example.com'
```
````

Consolidating as it's all part of setup